akd/src/tests.rs

```diff
@@ -8,16 +8,78 @@
 
 //! Contains the tests for the high-level API (directory, auditor, client)
 
+use std::collections::HashMap;
+
 use crate::{
     auditor::audit_verify,
     client::{key_history_verify, lookup_verify},
     directory::{Directory, PublishCorruption},
     ecvrf::{HardCodedAkdVRF, VRFKeyStorage},
-    errors::AkdError,
-    storage::{manager::StorageManager, memory::AsyncInMemoryDatabase, types::DbRecord, Database},
+    errors::{AkdError, StorageError},
+    storage::{
+        manager::StorageManager,
+        memory::AsyncInMemoryDatabase,
+        types::{DbRecord, KeyData, ValueState, ValueStateRetrievalFlag},
+        Database, DbSetState, Storable,
+    },
     AkdLabel, AkdValue, HistoryParams, HistoryVerificationParams, VerifyResult,
 };
 
+#[derive(Clone)]
+pub struct LocalDatabase;
+
+unsafe impl Send for LocalDatabase {}
+unsafe impl Sync for LocalDatabase {}
+
+mockall::mock! {
+    pub LocalDatabase {
+
+    }
```

Comment:
Nit: Any reason not to use `MockDatabase`? I find it a bit more apt than the "Local" convention.

Reply:
The "Mock" prepend is added by the `mock!` macro, so it'll be MockLocalDatabase. I didn't want to define a `Database` struct to risk clashing with the trait.

Reply:
Ahh, that's right! I totally forgot that the "mock" prefix is added with the `mock!` macro. I thought only the `#[automock]` attribute added it for some reason. Thanks!